lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -412,15 +418,27 @@ namespace {
       case MatchKind::NonObjC:
         return false;
       }
+
+      // Work around MSVC warning: not all control paths return a value
+      llvm_unreachable("All switch cases are covered");
     }
 
     SmallVector<Substitution, 2> WitnessSubstitutions;
 
     swift::Witness getWitness(ASTContext &ctx,
                               RequirementEnvironment &&reqEnvironment) const {
+
+      // C++ doesn't guarantee the function passed to param3 will be invoked
+      // before the function passed to param4. In Clang it is left to right,
+      // but not with MSVC.
+      // This causes an assertion compiling Swift code with an MSVC swift.exe
+      // Therefore, we  manually invoke getSyntheticEnvironment() before
+      // invoking takeRequirementToSyntheticMap()
```

Comment:
Eh, I don't think you even need this comment. We consider it bad style to depend on left-to-right-ness even when compiling with Clang. Thank you for finding and fixing this!